compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/staticcodeanalyzer/RuleFactory.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.http.compiler.staticcodeanalyzer;
+
+import io.ballerina.scan.Rule;
+import io.ballerina.scan.RuleKind;
+
+/**
+ * {@code RuleFactory} contains the logic to create a {@link Rule}.
+ */
+public class RuleFactory {
```

Comment:
Should this be a functionality provided by the ballerina scan package? IIUC we need to have `RuleFactory` and `RuleImpl` in other libraries as well

Reply:
The scanner package doesn't provide an API to register rules - this is handled through the `rules.json` file instead. While it's not necessary to have rule-related classes in the library package (we can simply report rules using IDs defined in `rules.json`), using hardcoded IDs reduces code readability. Therefore, I created a Rule interface to obtain rule information, which also helps with testing.

If this not necessary I can remove these classes and update the code with just hardcoded rule ID.

Reply:
I prefer this `rules.json` approach. It is more readable. Let's have it as it is. 
But the point is we have to duplicate these classes in the other modules. I hope that is fine for now, since we even have an approach to hard-code the values in the code as well